src/wasm_opt.rs

```diff
@@ -91,3 +47,68 @@ fn download_wasm_opt() -> Result<&'static Path> {
 
     WASM_OPT_PATH.as_deref().map_err(|err| anyhow!("{}", err))
 }
+
+pub struct WasmOpt {
+    optimization_level: u32,
+    shrink_level: u32,
+    debug_info: bool,
+}
+
+impl WasmOpt {
+    pub fn new(optimization_level: u32) -> Self {
+        Self {
+            optimization_level,
+            shrink_level: 0,
+            debug_info: false,
+        }
+    }
+
+    pub fn shrink(mut self, shrink_level: u32) -> Self {
+        self.shrink_level = shrink_level;
+        self
+    }
+
+    pub fn debug(mut self) -> Self {
+        self.debug_info = true;
+        self
+    }
+
+    pub fn optimize(&self, binary_path: impl AsRef<Path>) -> Result<()> {
```

Comment:
for chaining multiple files
```suggestion
    pub fn optimize(self, binary_path: impl AsRef<Path>) -> Result<Self> {
```